src/lib.rs

```diff
@@ -138,26 +223,30 @@ where
   zn_secondary: G2::Scalar,
   _p_c1: PhantomData<C1>,
   _p_c2: PhantomData<C2>,
+  a1: PhantomData<A1>,
+  a2: PhantomData<A2>,
+  num_steps: usize,
 }
 
-impl<G1, G2, C1, C2> RecursiveSNARK<G1, G2, C1, C2>
+impl<G1, G2, C1, C2, A1, A2> RecursiveSNARK<G1, G2, C1, C2, A1, A2>
 where
   G1: Group<Base = <G2 as Group>::Scalar>,
   G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar> + Clone,
-  C2: StepCircuit<G2::Scalar> + Clone,
+  C1: StepCircuit<G1::Scalar, A1> + Clone,
+  C2: StepCircuit<G2::Scalar, A2> + Clone,
+  A1: Arity<<G1 as Group>::Scalar>,
+  A2: Arity<<G2 as Group>::Scalar>,
 {
-  /// Create a new `RecursiveSNARK`
-  pub fn prove(
-    pp: &PublicParams<G1, G2, C1, C2>,
-    num_steps: usize,
+  /// Initialize a `RecursiveSNARK`.
+  pub fn init(
+    pp: &PublicParams<G1, G2, C1, C2, A1, A2>,
     z0_primary: G1::Scalar,
     z0_secondary: G2::Scalar,
+    z1_primary: G1::Scalar,
+    z1_secondary: G2::Scalar,
+    first_primary_circuit: &C1,
```

Comment:
Change to `first_circuit_primary` and `first_circuit_secondary` to rhyme with the naming style of other arguments?

Reply:
Changed.